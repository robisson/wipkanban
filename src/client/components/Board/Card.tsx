import * as React from "react";
import { withStyles, StyleRulesCallback } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import CardMedia from "@material-ui/core/CardMedia";
import Avatar from "@material-ui/core/Avatar";
import Chip from "@material-ui/core/Chip";
import Typography from "@material-ui/core/Typography";
import red from "@material-ui/core/colors/red";
import DateRangeIcon from "@material-ui/icons/DateRange";
import MessageIcon from "@material-ui/icons/Message";
import CheckBox from "@material-ui/icons/CheckBox";
import Attachment from "@material-ui/icons/Attachment";

const styles: StyleRulesCallback = () => ({
  card: {
    margin: 5,
    cursor: "pointer",
    width: 258
  },
  media: {
    height: 0,
    paddingTop: "56.25%" // 16:9
  },
  avatar: {
    width: 25,
    height: 25,
    float: "right",
    marginRight: 2
  },
  avatarDateRange: {
    width: 25,
    height: 25
  },
  CardContent: {
    padding: 2
  },
  chip: {
    margin: 0,
    backgroundColor: red[500],
    height: 20
  },
  chipDateRange: {
    margin: 2,
    height: 25
  }
});

type Props = {
  classes: any;
  title: string;
};

function CardBoard(props: Props) {
  const { classes, title } = props;
  return (
    <div className={classes.card}>
      <Card>
        <CardMedia
          className={classes.media}
          image="/images/contemplative-reptile.jpg"
          title="Contemplative Reptile"
        />
        <CardContent className={classes.CardContent}>
          <Chip className={classes.chip} />
          <Typography component="p">{title}</Typography>
          <Chip
            avatar={
              <Avatar className={classes.avatarDateRange}>
                {" "}
                <DateRangeIcon />{" "}
              </Avatar>
            }
            label="25/06/2017"
            className={classes.chipDateRange}
          />
          <Chip
            avatar={
              <Avatar className={classes.avatarDateRange}>
                {" "}
                <MessageIcon />{" "}
              </Avatar>
            }
            label="4"
            className={classes.chipDateRange}
          />
          <Chip
            avatar={
              <Avatar className={classes.avatarDateRange}>
                {" "}
                <CheckBox />{" "}
              </Avatar>
            }
            label="5/8"
            className={classes.chipDateRange}
          />
          <Chip
            avatar={
              <Avatar className={classes.avatarDateRange}>
                {" "}
                <Attachment />{" "}
              </Avatar>
            }
            label="5"
            className={classes.chipDateRange}
          />
          <Avatar
            className={classes.avatar}
            aria-label="Robisson Oliveira"
            src="/uploads/7b435896-aeca-4660-aaa3-b734e3e9200a.jpg"
          />
        </CardContent>
      </Card>
    </div>
  );
}

export default withStyles(styles)(CardBoard);
