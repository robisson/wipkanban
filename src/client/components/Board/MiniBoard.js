//@flow
import * as React from "react";
import { withStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import Typography from "@material-ui/core/Typography";
import CardActions from "@material-ui/core/CardActions";
import { Link } from "react-router-dom";

const styles = () => ({
  card: {
    display: "flex"
  },
  content: {
    display: "flex",
    flexWrap: "wrap",
    padding: 0
  },
  child: {
    flex: "1 0 30%",
    margin: "2px",
    height: "85px"
  },
  cardHeader: {
    padding: 5
  }
});

type Props = {
  classes: Object,
  _id: number,
  name: string,
  createdBy: string,
  createdAt: string,
  leadTime: number,
  cycleTime: number,
  throuput: number
};

function MediaControlCard(props: Props) {
  const {
    classes,
    _id,
    name,
  } = props;

  return (
    <Card>
      <div className={classes.cardHeader}>
        <Typography variant="h6">{name}</Typography>
      </div>
      <CardActions
        style={{
          display: "block",
          textAlign: "right"
        }}
      >
        <Link to={`/board/${_id}`} size="small" color="primary">
          Go to Board
        </Link>
      </CardActions>
    </Card>
  );
}

export default withStyles(styles)(MediaControlCard);
