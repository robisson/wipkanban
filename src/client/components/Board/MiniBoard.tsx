import * as React from "react";
import { withStyles, StyleRulesCallback } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import Typography from "@material-ui/core/Typography";
import CardActions from "@material-ui/core/CardActions";
import { Link } from "react-router-dom";

const styles: StyleRulesCallback = () => ({
  card: {
    display: "flex"
  },
  content: {
    display: "flex",
    flexWrap: "wrap",
    padding: 0
  },
  child: {
    flex: "1 0 30%",
    margin: "2px",
    height: "85px"
  },
  cardHeader: {
    padding: 5
  }
});

interface IMediaControlCardProps {
  classes: any;
  _id: number;
  name: string;
  createdBy: string;
  createdAt: string;
  leadTime: number;
  cycleTime: number;
  throuput: number;
}

function MediaControlCard(props: IMediaControlCardProps) {
  const { classes, _id, name } = props;

  return (
    <Card>
      <div className={classes.cardHeader}>
        <Typography variant="h6">{name}</Typography>
      </div>
      <CardActions
        style={{
          display: "block",
          textAlign: "right"
        }}
      >
        <Link to={`/board/${_id}`} color="primary">
          Go to Board
        </Link>
      </CardActions>
    </Card>
  );
}

export default withStyles(styles)(MediaControlCard);
