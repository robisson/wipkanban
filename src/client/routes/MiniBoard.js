//@flow
import * as React from 'react';
import {withStyles} from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import CardActions from '@material-ui/core/CardActions';
import Button from '@material-ui/core/Button';

const styles = () => ({
  card: {
    display: 'flex'
  },
  content: {
    display: 'flex',
    flexWrap: 'wrap',
    padding: 0
  },
  child: {
    flex: '1 0 30%',
    margin: '2px',
    height: '85px'
  },
  cardHeader: {
    padding: 5
  }
});

type Props = {
  classes: Object
}

function MediaControlCard(props : Props) {
  const {classes} = props;

  return (
    <Card>
      <div className={classes.cardHeader}>
        <Typography variant="title">
          Minhas Atividades
        </Typography>
        <Typography variant="subheading">
          createdb by you at today 12:00pm
        </Typography>
      </div>
      <CardContent className={classes.content}>
        <div className={classes.child}>
          <Typography variant="body2" color="primary" align="center">
            Lead Time
          </Typography>
          <Typography variant="display1" align="center">
            10
          </Typography>
          <Typography variant="body1" align="center">
            by week
          </Typography>
        </div>
        <div className={classes.child}>
          <Typography variant="body2" color="primary" align="center">
            Cycle Time
          </Typography>
          <Typography variant="display1" align="center">
            10
          </Typography>
          <Typography variant="body1" align="center">
            by week
          </Typography>
        </div>
        <div className={classes.child}>
          <Typography variant="body2" color="primary" align="center">
            Throghput
          </Typography>
          <Typography variant="display1" align="center">
            10
          </Typography>
          <Typography variant="body1" align="center">
            by week
          </Typography>
        </div>
      </CardContent>
      <CardActions
        style={{
        display: 'block',
        textAlign: 'right'
      }}>
        <Button size="small" color="primary">
          Go to Board
        </Button>
      </CardActions>
    </Card>
  );
}

export default withStyles(styles, {withTheme: true})(MediaControlCard);
