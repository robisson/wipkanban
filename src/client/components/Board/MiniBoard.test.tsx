import React from "react";
import MiniBoard from "./MiniBoard";
import { createShallow } from "@material-ui/core/test-utils";
import Card from "@material-ui/core/Card";
import { ShallowWrapper } from "enzyme";

describe("<MiniBoard /> Component", () => {
  let shallow:any;
  let wrapper:ShallowWrapper;

  beforeEach(() => {
    shallow = createShallow({ dive: true });
    wrapper = shallow(<MiniBoard _id={1} name="testsName" />);
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render a Card and CardContent component", () => {
    expect(wrapper.type()).toEqual(Card);
  });
});
