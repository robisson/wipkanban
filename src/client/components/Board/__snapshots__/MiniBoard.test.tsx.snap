// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MiniBoard /> Component should render correctly 1`] = `
<WithStyles(Card)>
  <div
    className="MediaControlCard-cardHeader-4"
  >
    <WithStyles(Typography)
      variant="h6"
    >
      testsName
    </WithStyles(Typography)>
  </div>
  <WithStyles(CardActions)
    style={
      Object {
        "display": "block",
        "textAlign": "right",
      }
    }
  >
    <Link
      color="primary"
      replace={false}
      to="/board/1"
    >
      Go to Board
    </Link>
  </WithStyles(CardActions)>
</WithStyles(Card)>
`;
